ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/layered/BaseFeePrioritizedTransactions.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.eth.transactions.layered;
+
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.eth.transactions.PendingTransaction;
+import org.hyperledger.besu.ethereum.eth.transactions.TransactionPoolConfiguration;
+import org.hyperledger.besu.ethereum.eth.transactions.TransactionPoolMetrics;
+import org.hyperledger.besu.ethereum.mainnet.feemarket.BaseFeeMarket;
+import org.hyperledger.besu.ethereum.mainnet.feemarket.FeeMarket;
+
+import java.util.Comparator;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Holds the current set of pending transactions with the ability to iterate them based on priority
+ * for mining or look-up by hash.
+ *
+ * <p>This class is safe for use across multiple threads.
+ */
+public class BaseFeePrioritizedTransactions extends AbstractPrioritizedTransactions {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BaseFeePrioritizedTransactions.class);
+  private Optional<Wei> nextBlockBaseFee;
+
+  public BaseFeePrioritizedTransactions(
+      final TransactionPoolConfiguration poolConfig,
+      final Supplier<BlockHeader> chainHeadHeaderSupplier,
+      final TransactionsLayer nextLayer,
+      final TransactionPoolMetrics metrics,
+      final BiFunction<PendingTransaction, PendingTransaction, Boolean>
+          transactionReplacementTester,
+      final BaseFeeMarket baseFeeMarket) {
+    super(poolConfig, nextLayer, metrics, transactionReplacementTester);
+    this.nextBlockBaseFee =
+        Optional.of(calculateNextBlockBaseFee(baseFeeMarket, chainHeadHeaderSupplier.get()));
+  }
+
+  @Override
+  protected int compareByFee(final PendingTransaction pt1, final PendingTransaction pt2) {
```

Comment:
this is a good point, I need to review and improve the prioritization of local txs, but would prefer to put them in a following PR, to avoid adding even more to this one

Reply:
ah ok, so isReceivedFromLocalSource is considered for the existing BaseFeePendingTransactionsSorter.
Maybe a TODO would make this clear if it's a missing feature?

Reply:
Will create tickets for the next things todo